doc/read.md

```diff
@@ -0,0 +1,128 @@
+
+#### `READ` client set command
+
+    READ <target> <timestamp>
+
+When sent from a client, this `READ` command signals to the server that the last message read by the user, to the best knowledge of the client, has the specified timestamp. The timestamp MUST correspond to a previous message `time` tag. The timestamp MUST NOT be a literal `*`.
+
+The server MUST reply to a successful `READ` set command using a `READ` server command, or using an error message.
+
+#### `READ` client get command
+
+    READ <target>
+
+When sent from a client, this `READ` command requests the server about the timestamp of the last message read by the user.
+
+The server MUST reply to a successful `READ` get command using a `READ` server command, or using an error message.
+
+#### `READ` server command
+
+When sent from a server, the `READ` command signals to the client that the last message read by the user, to the best knowledge of the server, has the specified timestamp. In that case, the command has the following syntax:
+
+    <prefix> READ <target> <timestamp>
+
+The `prefix` is the prefix of the client the message is sent to.
+
+If there is no known last message read timestamp, the `timestamp` parameter is a literal `*`. Otherwise, it is the formatted timestamp of the last read message.
+
+#### Command flows
+
+The server sends a `READ` command to a client in the following cases.
+
+If the `soju.im/read` capability is negotiated, after the server sends a server `JOIN` command to the client for a corresponding channel, the server MUST send a `READ` command for that channel. The command MUST be sent before the `RPL_ENDOFNAMES` reply for that channel following the `JOIN`.
+
+If the `soju.im/read` capability is negotiated, after the last read timestamp of a target changes, the server SHOULD send a `READ` command for that target to all the clients of the user.
+
+#### Read timestamp notes
+
+The last read timestamp of a target SHOULD only ever increase. If a client sends a `READ` command with a timestamp that is below or equal to the current known timestamp of the server, the server SHOULD reply with a `READ` command with the newer, previous value that was stored and ignore the client timestamp.
+
+#### Errors and Warnings
+
+Errors are returned using the standard replies syntax.
+
+If the server receives a `READ` command with missing parameters, the `NEED_MORE_PARAMS` error code MUST be returned.
+
+    FAIL READ NEED_MORE_PARAMS :Missing parameters
+
+If the selectors or limit supplied were invalid, the `INVALID_PARAMS` error code SHOULD be returned.
+
+    FAIL READ INVALID_PARAMS [invalid_parameters] :Invalid parameters
+
+If the read timestamp cannot be set or returned due to an error, the `MESSAGE_ERROR` error code SHOULD be returned.
```

Comment:
Nit: I think chathistory has been using the wording `MESSAGE_ERROR` as a way to indicate that the message store hit an error. Here no message store is involved, so maybe should rename this to something like `TEMPORARILY_UNAVAILABLE` (like the account-registration spec), `READ_ERROR` or `INTERNAL_ERROR`.